src/registry.ts

```diff
@@ -130,8 +130,21 @@ function headOk(url: string, headers: Headers): Promise<boolean> {
 		options.method = "HEAD";
 		request(options, (res) => {
 			logger.debug(`HEAD ${url}`, res.statusCode);
+			// Not found
 			if (res.statusCode == 404) return resolve(false);
+			// OK
 			if (res.statusCode == 200) return resolve(true);
+			// Redirect. We assume that the layer exists at the redirect location
+			if (res.statusCode == 301 || res.statusCode == 307 || res.statusCode == 308) return resolve(true);
+			// Moved Permanently
+			if (res.statusCode == 301 && res.headers.location) return resolve(headOk(res.headers.location, headers));
```

Comment:
Wouldn't these three next `if`s be caught by the previous `if` ?

Reply:
Yes, they would. I should've been more explicit about this being work in progress.

Reply:
Yeah, this one is on me. Should have seen the "wip" in the title.